### DISCUSSION TITLE
TGE 的發送流程

### DISCUSSION THREAD

**Discussion by yurenju** [lun-dao/LunDAO#49] [General]
如 #11 的討論，我們先暫時決定在 Arbitrum 運作，如果已經決定了採用哪個 Layer 2，我想要把部分 TGE 的發送直接搬到 Arbitrum。想先說一下要把 LUN Bridge 到 Arbitrum 要怎麼做。

根據 [Offchainlabs 的文件][1] 如果一個 ERC20 token 還沒 bridge 過，我們會需要跑一個 Script 來 deploy L2 token 到 arbitrum 去。從我們的狀況來看就是要有人先跑這個 script 建立 bridge 通道，剩下的 token 就可以用 [Arbitrum Token Bridge][2] 送到 Arbitrum 了。

基於以上的狀況的話，我想提出 TGE 的發送流程是這樣：
1. 建立一個 arbitrum 的 gnosis safe
2. 把所有的 LUN 都 bridge 到 arbitrum 的 gnosis 裡面
3.  依照比例用 [Disperse.app][3] 依照比例發送 L2 LUN

雖然這樣之前建立的 ehtereum mainnet gnosis safe 有點脫褲子放屁，但我覺得有 L1 的基礎建設還是必要的啦 🤣
大家覺得這樣的流程如何？

[1]: https://developer.offchainlabs.com/docs/bridging_assets#default-standard-bridging
[2]: https://bridge.arbitrum.io/
[3]: https://disperse.app/

**Comment by KimiWu123** [comment]
恩，我這兩天也在想這個問題。不過有一點我不太清楚，就是LUN要怎麼兌現？跟 DAO 換 ETH 出來嗎？如果是這樣，那 DAO 移到 Arb 比較合理，不然轉到 mainnet 在做交換，太不合成本了 😆 。如果我上述的流程都對，那我會建議整組（DAO, multisig, 兌現）搬到 Arb 上

**Reply by yurenju** [reply]
對，我是覺得整組機制都搬到 arbitrum 上，這樣的話兩邊各會有：
- ethereum mainnet: gnosis safe, L1 LUN 
- arbitrum: gnosis safe, L2 LUN, Disperse.app, coordinape (這個我會先試一下), Uniswap liquidity

我目前想說是把現有的 ETH 跟 LUN 組成交易對放在 Arbitrum 的 Uniswap 讓大家來換錢。至於 mainnet 上先不放流動性了。當然就跟 #38 討論的一樣，如果沒有其他可以用 LUN 代幣的場景，隨著大家賣代幣，很快就會跌成 💩 了，我一直沒很確定這樣流動性較低的社群代幣到底怎麼做是比較好的方法，這部分就在 #38 討論，或是額外開個新的主題討論吧。

**Reply by iisaint** [reply]
建議可以先不要提供LUN/ETH 流動性，等到時機成熟再提供，例如達成下列條件之後再開放

- 文章累積至 23 篇
- 提供 **單向** 兌換機制 (online or offline 都行)，讓使用者用 ETH 來換 LUN
- 完成 13 次 Requests for Articles (RFA) 徵文
- 其他等等

**Reply by yurenju** [reply]
OK 也好，不過如果有其他人要做 RFA 那些 LUN token 要怎麼來？

**Reply by hydai** [reply]
走單向兌換機制如何，有點像是放更長時間的 TGE? 只是價格會先被 FIX 著？

**Reply by iisaint** [reply]
我們可以提供 單向兌換機制 讓有需要的人可以來換

**Reply by yurenju** [reply]
ＯＫＯＫ，那我們就先這麼做好了，跑一陣子再回頭來看看要怎麼調整作法

**Reply by yurenju** [reply]
補充一下其實我還是比較偏好先提供流動性，因為畢竟 LUN 對於作者而言是獎勵性質，可以有地方兌換會比較好，沒地方兌換就稱不上獎勵性質了，提供流動性同時可以解決作者可以兌現，而且想要買 LUN 的人可以直接去 Uniswap 買。

不過我們還是可以先跑一陣子試試看，看情況再來放流動性。
